pkgs/development/python-modules/semgrep/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib, buildPythonPackage, runCommand, fetchPypi, fetchurl, jsonschema, wcmatch
+, click, ruamel-yaml, click-option-group, boltons, tqdm, glom, requests, colorama
+, defusedxml, packaging, peewee, setuptools }:
+
+let
+  version = "0.92.0";
+
+  # semgrep engine is written in OCaml and sources are present in
+  # https://github.com/returntocorp/semgrep, but I failed to build it.
+  #
+  # It looks like dypgen required ocaml <= 4.06 and atdgen ocaml >= 4.08.
+  #
+  # Clearly, I don't understand something, and this can and should be done
+  # properly. In mean time, I cheat and download upstream-provided static
+  # binary. That constraints derivation to x86_64 architecture.
+  #
+  # 2022-05-12 ~kaction
+  semgrep-core = let
+    env = {
+      src = fetchurl {
+        url =
+          "https://github.com/returntocorp/semgrep/releases/download/v${version}/semgrep-v${version}-ubuntu-16.04.tgz";
+        sha256 = "0ba488b41qyh0n1svzjxln2ln9n0x3njysq7qkxcz7dw5raz5l36";
+      };
+    };
+  in runCommand "semgrep-core" env ''
+    mkdir -p $out/bin
+    tar xfv $src
+    cp ./semgrep-files/semgrep-core $out/bin
+  '';
+
+in buildPythonPackage rec {
+  pname = "semgrep";
+  inherit version;
+
+  # FIXME: This is also cheating, should fetch from github.
```

Comment:
We should do this right now.

Reply:
That involves compiling ocaml. Will take indefinite amount of time.

Reply:
You sure that that perfect is better than now?

Reply:
And, tbh, we have a lot of packages that build from pypi, not from git.

Reply:
> That involves compiling ocaml. Will take indefinite amount of time.

We have ocaml already in nixpkgs with quite some packages and versions of the compiler. If upstream has genenric build scripts it should ™️ be quite easy to compile it.

> we have a lot of packages that build from pypi

Those are mostly using dist tarballs without any compiled source.

Reply:
Well, I have very vague idea about what constitutes standard build package, but so far:

```
  # It looks like dypgen required ocaml <= 4.06 and atdgen ocaml >= 4.08.
```

Reply:
Can you include your attempt at building it? I was working on this in parallel and kept running into missing dependencies.